### PR TITLE
main,render: replace assert(false) with abort().

### PR DESCRIPTION
**Pull request by ericonr**
Also add error message about lack of JPEG support.

### CODE REVIEW

File: main.c [hunk 1]
```diff
@@ -216,7 +216,8 @@ bool default_filename(char *filename, size_t n, int filetype) {
 		ext = "jpeg";
 		break;
 #else
-		assert(false);
+		fprintf(stderr, "this version of grim doesn't support JPEG\n");
```
Comment (emersion) [inline main.c:219]: No need for pretty error messages, we already have a user-friendly message on line 324.

Comment (ericonr) [reply main.c:219]: Oh, got it!

Comment (ericonr) [reply main.c]: Done

**Review by emersion — APPROVED** [review]
Thanks!

**Pull request merged**